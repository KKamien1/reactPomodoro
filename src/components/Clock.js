import React from 'react';

function Clock({ hours = 34, minutes = -77, seconds = -77, miliseconds = 3 }) {
    const format = (digits = 2, max = 59, placeholder = "0") => num => Math.min(Math.max(0, num), max).toString().padStart(digits, placeholder);

    const to24 = format(2, 23);
    const to60 = format();          // default setup for minutes and seconds
    const to1000 = format(3, 999);

    const h = to24(hours);
    const ms = to1000(miliseconds);
    const [m, s] = [minutes, seconds].map(to60);

    return (<h2 className="Clock">Pozosta≈Ço {h}:{m}:{s}.{ms}</h2>);
}

export default Clock;