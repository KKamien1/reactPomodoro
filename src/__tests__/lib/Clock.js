import React from 'react';
import Clock from '../../components/Clock';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'


describe('<Clock> component on DOM', () => {
    let root;
    beforeEach(() => {
        root = document.createElement("div");
        ReactDOM.render(<Clock minutes={10} seconds={20} />, root)
    })

    it('renders properly H2 element', () => {
        expect(root.childNodes[0].nodeName).toEqual("H2")
    });
    it('renders properly with className', () => {
        expect(root.childNodes[0].className).toEqual("clock")
    });
    it('renders properly with text', () => {
        expect(root.childNodes[0].textContent).toEqual("Pozostało 00:10:20.000")
    });
    it('sets ', () => {
        expect(<Clock />).toEqual(<Clock hours={0} miliseconds={0} minutes={20} seconds={0} />);
    });
})

describe('<Clock> component with react-test-renderer', () => {
    let clockRenderer;
    beforeEach(() => {
        clockRenderer = renderer.create(<Clock minutes={10} seconds={20} />)
    })

    it('renders properly', () => {
        expect(clockRenderer.toJSON()).toMatchSnapshot()
    });
    it('renders properly with h2 element', () => {
        expect(clockRenderer.toJSON().type).toEqual('h2')
    });
    it('renders properly with className', () => {
        expect(clockRenderer.toJSON().props).toMatchObject({ "className": expect.stringMatching(/clock/) })
    });
    xit('renders properly with text', () => {
        expect(clockRenderer.toJSON().children).toEqual(['Pozostało ', '00', ':', <span className="clock__min">10</span>, <span className="clock__sepator">:</span>, <span className="clock__sec">20</span>, '.', '000'])
    });
})

