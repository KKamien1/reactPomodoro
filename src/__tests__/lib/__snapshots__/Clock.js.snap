// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock> component with react-test-renderer renders properly 1`] = `
<h2
  className="clock"
>
  Pozosta≈Ço 
  00
  :
  <span
    className="clock__min"
  >
    10
  </span>
  <span
    className="clock__sepator"
  >
    :
  </span>
  <span
    className="clock__sec"
  >
    20
  </span>
  .
  000
</h2>
`;
